AbletonMCP_Remote_Script/__init__.py

```
@@ -799,8 +1015,1196 @@ def _find_browser_item_by_uri(self, browser_or_item, uri, max_depth=10, current_
+                "new_index": new_track_index,
+                "new_name": new_track.name
+            }
+        except Exception as e:
+            self.log_message("Error duplicating track: " + str(e))
+            raise
+
+    def _quantize_clip_notes(self, track_index, clip_index, grid_size):
+        """Quantize MIDI notes in a clip to a grid"""
+        try:
+            if track_index < 0 or track_index >= len(self._song.tracks):
+                raise IndexError("Track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)